route_linux.go

```diff
@@ -186,6 +186,71 @@ func (e *MPLSEncap) Equal(x Encap) bool {
 	return true
 }
 
+// SEG6 definitions
+type SEG6Encap struct {
+	Mode int
+	Srh  nl.IPv6SrHdr
```

Comment:
This is the last thing I am not convinced about.

`Seg6Encap` is `netlink` pkg user interface, lib client will have to provide it when calling `netlink.AddRoute()`.
I am not sure we should have client import `netlink/nl` for setting `Srh`, as it is done now for convenience I guess, where only  its `Segments` field is exported.

I took a look at other `netlink` pkg API, for the most no `netlink/nl` types need to be fed in.

Probably this type could be changed to 
```
type SEG6Encap struct {
	Mode         int
	Segments  []net.IP
}
``` 

and the respective `nl.IPv6SrHdr` could be created internally when interfacing with `netlink/nl` methods.

Reply:
I have changed to `Segments  []net.IP` as you suggested.
I removed using `IPv6SrHdr` in `EncodeSEG6Encap` to make code cleaner and short.
I kept using it in `DecodeSEG6Encap` in case one wants to do granular check with `func (s1 *IPv6SrHdr) Equal(s2 IPv6SrHdr) bool` by directly using netlink/nl.
For example, there could be a case that only `segmentsLeft` `firstSegment` are different for two seg6 route entries. However, explicitly configuring those values via iproute2 is not currently supported. Thus, will leave Encap part for future when someone actually wants to do such operation.